lib/src/slices/data_bkg/data_bkg_service.dart

```diff
@@ -13,265 +12,156 @@ import '../api_app_data/model/api_app_data_model.dart';
-          format: "metadata",
-          headers: ["List-unsubscribe"]);
-      if (message?.sender?.email != null &&
-          message?.sender?.unsubscribeMailTo != null) {
-        messages.add(message!);
-        if (message.receivedDate != null &&
-            message.receivedDate!.isBefore(first))
-          first = message.receivedDate!;
+  Future<void> fetchData(ApiAuthServiceAccountModel account) async {
+    DataBkgServiceProviderInterface? provider = getProvider(account);
+    if (provider != null) {
+      _log.fine("fetch data for " + provider.account.provider!);
+      if (provider is DataBkgServiceEmailInterface) {
+        _log.fine("fetch email data for " + provider.account.email!);
+        await _fetchEmail(provider);
       }
-    }
-    ApiEmailSenderModel sender = messages.first.sender!;
-    sender.emailSince = first;
-    ApiEmailSenderModel? inserted = await _saveSender(sender);
-    if (inserted != null) {
-      for (ApiEmailMsgModel message in messages) {
-        message.sender = inserted;
-        await _apiEmailMsgService.upsert(message);
-      }
-      _log.fine('Sender upsert: ' + (sender.company?.domain ?? ''));
       notifyListeners();
     }
-    return messages;
   }
 
-  Future<List<String>> _checkGmailNewSenderPage(
-      {required String email,
-      String? page,
-      required List<String> messages}) async {
-    DataBkgModelPage<String> res =
-        await _apiGoogleService.gmailFetch(query: 'from:' + email, page: page);
-    if (res.data != null) messages.addAll(res.data!);
-    if (res.next != null)
-      return _checkGmailNewSenderPage(
-          email: email, page: res.next, messages: messages);
-    else
-      return messages;
+  void addAccount(ApiAuthServiceAccountModel account) {
+    _accounts.add(account);
+    fetchData(account);
   }
 
-  Future<ApiEmailSenderModel?> _saveSender(ApiEmailSenderModel sender) async {
-    if (sender.email != null) {
-      ApiCompanyModelLocal? company =
-          await _apiCompanyService.upsert(domainFromEmail(sender.email!));
-      if (company != null) {
-        sender.company = company;
-        ApiEmailSenderModel saved = await _apiEmailSenderService.upsert(sender);
-        return saved;
-      }
-    }
+  Future<void> _fetchEmail(DataBkgServiceProviderInterface provider) async {
+    DataBkgServiceEmail dataBkgServiceEmail = DataBkgServiceEmail(
+        this._apiAuthService,
+        provider as DataBkgServiceEmailInterface,
+        this._apiCompanyService,
+        this._apiEmailSenderService,
+        this._apiEmailMsgService,
+        this._apiAppDataService);
+    await dataBkgServiceEmail.checkEmail();
   }
 
-  String domainFromEmail(String email) {
-    List<String> atSplit = email.split('@');
-    List<String> periodSplit = atSplit[atSplit.length - 1].split('.');
-    return periodSplit[periodSplit.length - 2] +
-        "." +
-        periodSplit[periodSplit.length - 1];
+  DataBkgServiceProviderInterface? getProvider(
```

Comment:
Should return a list of provider interfaces. 

The standard provider interface should have an abstract method like getServices so ApiGoogleService should implement getServices which would return [ApiGoogleServiceEmail]